server/event_handler.py

```diff
@@ -103,23 +99,22 @@ async def __call__(self, request: EventRequest, room_code: str) -> None:
                 )
                 await self.manager.broadcast(response, room_code, sender=self.client)
 
-                WebSocketDisconnect.response = response  # type: ignore
-                raise WebSocketDisconnect
+                self.manager.disconnect(self.client, room_code)
             case EventType.MOVE:
                 move_data = cast(MoveData, event_data)
                 self.room.cursors[self.client.id] = move_data.position
 
                 # Broadcast to every client a move event to update the cursors'
                 # positions
                 response = EventResponse(type=EventType.MOVE, data=move_data, status_code=StatusCode.SUCCESS)
-                await self.manager.broadcast(response, room_code)
+                await self.manager.broadcast(response, room_code, sender=self.client)
             case EventType.REPLACE:
                 replace_data = cast(ReplaceData, event_data)
                 self.room.update_code(replace_data)
 
                 # Broadcast to every client a replace event to update the code
                 response = EventResponse(type=EventType.REPLACE, data=replace_data, status_code=StatusCode.SUCCESS)
-                await self.manager.broadcast(response, room_code)
+                await self.manager.broadcast(response, room_code, sender=self.client)
```

Comment:
Same as comment above

Reply:
The replace shouldn't be broadcasted to the client as well because if the client edits the code and then sends the replace event, his code will be already modified.